task8/ProfitApp/src/main/java/com/bsu/fpmi/ProfitApp/AdCollection.java

```diff
@@ -0,0 +1,180 @@
+package com.bsu.fpmi.ProfitApp;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AdCollection {
+
+    List<Ad> ads;
+
+    AdCollection() {
+        ads = new ArrayList<>();
+    }
+
+    public Ad get(String id) {
+        return ads.stream().filter(a -> a.getId().equals(id)).findAny().orElse(null);
+    }
+
+    public boolean remove(String id) {
+        return ads.removeIf(a -> a.getId().equals(id));
+    }
+
+    public List<Ad> getPage(int skip, int top, FilterConfig fc, String sortType) {
+        List<Ad> toReturn = ads;
+        SimpleDateFormat df = new SimpleDateFormat("dd.MM.yyyy");
+        if (sortType.equalsIgnoreCase("Rating")) {
+            toReturn = toReturn.stream().sorted((a, b) -> Double.compare(b.getRating(), a.getRating())).collect(Collectors.toList());
+        } else {
+            toReturn = toReturn.stream().sorted((a, b) ->
+                    {
+                        try {
+                            return df.parse(b.getCreatedAt()).compareTo(df.parse(a.getCreatedAt()));
```

Comment:
That's why storing a date at once would give you a more brief notation. Some default date could be stored if the provided value is unparseable. Then there would be no nested `{}` at all